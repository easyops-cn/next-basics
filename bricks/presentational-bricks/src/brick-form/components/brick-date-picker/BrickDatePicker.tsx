import React from "react";
import { DatePicker } from "antd";
import { WrapperFormItem } from "../wrapper-form-item/WrapperFormItem";
import { handleChange } from "../../processor";
import {
  DatePickerProps,
  MonthPickerProps,
  RangePickerProps,
  WeekPickerProps
} from "antd/lib/date-picker/interface";

export interface LegacyBrickDatePickerProps {
  componentType?: "DatePicker" | "MonthPicker" | "RangePicker" | "WeekPicker";
  configProps?:
    | DatePickerProps
    | MonthPickerProps
    | RangePickerProps
    | WeekPickerProps;
  onChange: Function;
}

export function LegacyBrickDatePicker({
  componentType = "DatePicker",
  configProps,
  onChange
}: LegacyBrickDatePickerProps): React.ReactElement {
  let element: React.ReactElement;

  const onHandleChange = (e: any) => handleChange(e, onChange);

  switch (componentType) {
    case "MonthPicker":
      element = (
        <DatePicker.MonthPicker
          {...(configProps as MonthPickerProps)}
          onChange={onHandleChange}
        />
      );
      break;
    case "RangePicker":
      element = (
        <DatePicker.RangePicker
          {...(configProps as RangePickerProps)}
          onChange={onHandleChange}
        />
      );
      break;
    case "WeekPicker":
      element = (
        <DatePicker.WeekPicker
          {...(configProps as WeekPickerProps)}
          onChange={onHandleChange}
        />
      );
      break;
    default:
      element = (
        <DatePicker
          {...(configProps as DatePickerProps)}
          onChange={onHandleChange}
        />
      );
  }
  return element;
}

export const BrickDatePicker = WrapperFormItem(LegacyBrickDatePicker);
