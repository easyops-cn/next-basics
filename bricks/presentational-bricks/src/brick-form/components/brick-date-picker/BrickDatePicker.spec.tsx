import React from "react";
import { shallow } from "enzyme";
import { LegacyBrickDatePicker } from "./BrickDatePicker";

const onChange = jest.fn();

describe("BrickDatePicker", () => {
  it("should work when componentType undefined", () => {
    const wrapper = shallow(<LegacyBrickDatePicker onChange={onChange} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should work when componentType is 'MonthPicker'", () => {
    const wrapper = shallow(
      <LegacyBrickDatePicker componentType="MonthPicker" onChange={onChange} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should work when componentType is 'RangePicker'", () => {
    const wrapper = shallow(
      <LegacyBrickDatePicker componentType="RangePicker" onChange={onChange} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should work when componentType is 'WeekPicker'", () => {
    const wrapper = shallow(
      <LegacyBrickDatePicker componentType="WeekPicker" onChange={onChange} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("should trigger change event", () => {
    const wrapper = shallow(<LegacyBrickDatePicker onChange={onChange} />);
    wrapper.simulate("change");
    expect(onChange).toHaveBeenCalled();
  });
});
