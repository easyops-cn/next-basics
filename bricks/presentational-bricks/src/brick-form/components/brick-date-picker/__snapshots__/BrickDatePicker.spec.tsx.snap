// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrickDatePicker should work when componentType is 'MonthPicker' 1`] = `
<MonthPicker
  onChange={[Function]}
/>
`;

exports[`BrickDatePicker should work when componentType is 'RangePicker' 1`] = `
<RangePicker
  onChange={[Function]}
/>
`;

exports[`BrickDatePicker should work when componentType is 'WeekPicker' 1`] = `
<WeekPicker
  onChange={[Function]}
/>
`;

exports[`BrickDatePicker should work when componentType undefined 1`] = `
<Picker
  onChange={[Function]}
/>
`;
